Framework/Algorithms/src/RemoveExpDecay.cpp

```diff
@@ -100,15 +102,24 @@ void MuonRemoveExpDecay::exec() {
       throw std::invalid_argument(
           "Spectra size greater than the number of spectra!");
     }
-
-    // Remove decay from Y and E
-    outputWS->setHistogram(specNum, removeDecay(inputWS->histogram(specNum)));
-
-    // do scaling and subtract 1
-    const double normConst = calNormalisationConst(outputWS, spectra[i]);
-    outputWS->mutableY(specNum) /= normConst;
-    outputWS->mutableY(specNum) -= 1.0;
-    outputWS->mutableE(specNum) /= normConst;
+    auto fsdaf = (inputWS->y(specNum));
+    auto emptySpectrum = std::all_of(inputWS->y(specNum).begin(),
```

Comment:
Similar comment regarding the lambda and also as you have got the `fsdaf` variable then you could use that in `std::all_of` to avoid calling `y` repeatedly,